### PR TITLE
chore(oauth): remove modifications to OpenShift api-server

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Related to https://github.com/cryostatio/cryostat/issues/1489
Depends on https://github.com/cryostatio/cryostat-web/pull/1059

## Description of the change:

Remove the reconciler logics that modify the api-server to add cryostat application URL as allowed CORS origins.

## Motivation for the change:

https://github.com/cryostatio/cryostat-web/pull/1059 will submit form on the web client to clear login session. This removes the requirement for making CORS request.

### CODE REVIEW

**Comment by tthvo** [comment]
I think the main issue is that the application URL is not cleaned up after upgrades to newer operator. Wonder if we could do something about it or just document it for customers?

**Review by ebaron — COMMENTED** [review]
Hi @tthvo, thanks for handling this. I just have one suggestion: we should keep `deleteCorsAllowedOrigins`. This ensures that our modifications get removed when the user upgrades the operator.

**Comment by tthvo** [comment]
Hi I updated to only remove the addition logic and also updated tests. Any other tests you think we should add here?

**Comment by ebaron** [comment]
> Hi I updated to only remove the addition logic and also updated tests. Any other tests you think we should add here?

Hey Thuan, that's a good point. I think a test that verifies that an existing API server modification actually gets removed would be useful. We'd have to modify the APIServer manually as part of the test setup. Then the test itself could be like: https://github.com/cryostatio/cryostat-operator/blob/1a02ba82282cec37931b9993970dc416c0e1d08a/internal/controllers/reconciler_test.go#L1318-L1324

I also just realized that we should probably call `deleteCorsAllowedOrigins` from `reconcileOpenShift` and not `finalizeOpenShift`. We don't want the user to have to delete the CR for our modifications to be removed.

**Comment by ebaron** [comment]
> I also just realized that we should probably call `deleteCorsAllowedOrigins` from `reconcileOpenShift` and not `finalizeOpenShift`. We don't want the user to have to delete the CR for our modifications to be removed.

I guess it should actually be called from both methods. If for some reason a user upgraded the operator with a deleted Cryostat, we would need to make the modification during the finalizer step.

**Comment by tthvo** [comment]
Right! Thanks for the suggestions. I called `deleteCorsAllowedOrigins` in both functions now and added a new test to check if an existing application URL is removed.

File: internal/controllers/reconciler_test.go [hunk 1]
```diff
@@ -1282,8 +1282,20 @@ func (c *controllerTest) commonTests() {
...
+					t.objs = []ctrlclient.Object{
+						t.NewApiServerWithApplicationURL(),
+						t.NewNamespace(),
+					}
+				})
+				It("should remove the application url", func() {
+					apiServer := &configv1.APIServer{}
+					err := t.Client.Get(context.Background(), types.NamespacedName{Name: "cluster"}, apiServer)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(apiServer.Spec.AdditionalCORSAllowedOrigins).ToNot(ContainElement("https://cryostat\\.example\\.com"))
```
Comment (ebaron) [inline internal/controllers/reconciler_test.go:1296]: I would use `t.Name` here instead of `cryostat`.

**Review by ebaron — COMMENTED** [review]
Hey @tthvo, looks good! Just going to give it a try in OpenShift

File: internal/controllers/reconciler_test.go (hunk 1 above)
Comment (tthvo) [reply internal/controllers/reconciler_test.go]: Thanks! Updated now!

**Review by ebaron — COMMENTED** [review]
While testing, I noticed we get a lot of `Removed from APIServer CORS allowed origins` messages in the log. In the majority of cases there will be nothing to remove and I'd rather not have us interact with this config at all, even for a no-op.

Could I trouble you to just wrap the `client.Update` and log message in a conditional that only runs when a match is found?

https://github.com/cryostatio/cryostat-operator/blob/1a02ba82282cec37931b9993970dc416c0e1d08a/internal/controllers/openshift.go#L189-L204

**Comment by tthvo** [comment]
> While testing, I noticed we get a lot of Removed from APIServer CORS allowed origins messages in the log. In the majority of cases there will be nothing to remove and I'd rather not have us interact with this config at all, even for a no-op

Oh yess! I wonder so. Make sense to me! Updated now!

**Review by ebaron — APPROVED** [review]
Thanks @tthvo!

**Pull request merged**